### PR TITLE
Add experiment for removing facia front show more button

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

Add experiment for removing front show more button

## What is the value of this and can you measure success?

Provide us with information help make the decision for removing the show more button on fronts

## Tested in CODE?

No

### CODE REVIEW

File: common/app/layout/Front.scala [hunk 1]
```diff
@@ -365,10 +367,15 @@ object Front extends implicits.Collections {
     )
   }
 
+  private def applyShowMoreExperiment(faciaContainer: FaciaContainer)(implicit request: RequestHeader): FaciaContainer = {
+    val showMoreEnabled = ActiveExperiments.inOrOut(ShowMoreExperiment, in = false, out = true)
```
Comment (TBonnin) [inline common/app/layout/Front.scala]: could have been a simple 😀 
```
val showMoreEnabled = !ActiveExperiments.isParticipating(ShowMoreExperiment)
```

File: common/app/experiments/Experiments.scala [hunk 2]
```diff
@@ -61,3 +62,10 @@ object Garnett extends Experiment(
   participationGroup= Perc0C
 )
 
+object ShowMoreExperiment extends Experiment(
```
Comment (TBonnin) [inline common/app/experiments/Experiments.scala]: Should we name it `HideShowMoreButtonExperiment` to describe more explicitely what it does?

**Review by TBonnin — COMMENTED** [review]
Can we add the data-link info in the same PR? That would be easier to collect info about who has clicked on those show more stories?

**Comment by QuarpT** [comment]
I'll add the data-link-name changes to this PR

File: common/app/experiments/Experiments.scala (hunk 2 above)
Comment (QuarpT) [reply common/app/experiments/Experiments.scala]: I'll change it

File: common/app/layout/Front.scala (hunk 1 above)
Comment (QuarpT) [reply common/app/layout/Front.scala]: True. 
I'd still like to add `inOrOut` since inlining if statements for non-boolean experiments isn't so nice :P

Comment (TBonnin) [reply common/app/layout/Front.scala]: it is still `val myVal = if(ActiveExperiments.isParticipating(ShowMoreExperiment)) something else somethingElse`. Not really complex

**Comment by paperboyo** [comment]
Just an opinion: not that I care much for this particular device which always caused grief on its own (yank, misplacement because of js dependency) and presented boring, indigestible flurry of in-distinctive cards, but in general – I’m for long:


[ideally – as long as there are stories to show from backfill]

File: common/app/layout/Front.scala (hunk 1 above)
Comment (QuarpT) [reply common/app/layout/Front.scala]: 👍

File: common/app/layout/Front.scala [hunk 3]
```diff
@@ -365,10 +367,15 @@ object Front extends implicits.Collections {
     )
   }
 
+  private def applyShowMoreExperiment(faciaContainer: FaciaContainer)(implicit request: RequestHeader): FaciaContainer = {
+    val showMoreEnabled = !ActiveExperiments.isParticipating(HideShowMoreButtonExperiment)
+    faciaContainer.copy(hasShowMoreEnabled = showMoreEnabled)
+  }
+
   def fromPressedPage(pressedPage: PressedPage,
```
Comment (TBonnin) [inline common/app/layout/Front.scala]: unrelated to this PR but those `fromPressedPage...` function should be move to Facia which is the only placed they are called from

File: common/app/layout/Sublink.scala [hunk 4]
```diff
@@ -325,6 +326,12 @@ case class ContentCard(
     case _ => false
   }
 
+  def dataLinkName(index: Int): String = {
+    val name = s"$analyticsPrefix | card-@${index + 1}"
+    val withFromShowMore = if (fromShowMore) s"showmore | $name" else name
```
Comment (TBonnin) [inline common/app/layout/Sublink.scala]: I don't know anything about `data-link`. Is `showmore` supposed to be added as a prefix or appended at the end?

File: facia/app/controllers/FaciaController.scala [hunk 5]
```diff
@@ -190,17 +190,25 @@ trait FaciaController extends BaseController with Logging with ImplicitControlle
     }
   }
 
+  private def withFromShowMore(faciaCardAndIndex: FaciaCardAndIndex): FaciaCardAndIndex = faciaCardAndIndex.item match {
+    case contentCard: ContentCard => faciaCardAndIndex.copy(item = contentCard.copy(fromShowMore = true))
+    case _ => faciaCardAndIndex
+  }
```
Comment (TBonnin) [inline facia/app/controllers/FaciaController.scala]: Can this be moved in `FaciaCardAndIndex` class?

Comment (QuarpT) [reply facia/app/controllers/FaciaController.scala]: yes!

File: common/app/layout/Front.scala (hunk 3 above)
Comment (QuarpT) [reply common/app/layout/Front.scala]: Nice spot. I'll do this refactor in a separate PR

**Pull request merged**